keps/sig-network/2091-cluster-network-policy/README.md

```diff
@@ -0,0 +1,1161 @@
+### Test Plan
+
+- Add e2e tests for ClusterNetworkPolicy resource
+  - Ensure `Pass` rules are delegated and are not subject to CNP rules.
+  - Ensure `Deny` rules drop traffic.
+  - Ensure `Allow` rules allow traffic.
+  - Ensure that in stacked ClusterNetworkPolicies/K8s NetworkPolicies, precedence is maintained
+    as per the `priority` set in CNP.
+- e2e test cases must cover ingress and egress rules
+- e2e test cases must cover port-ranges, named ports, integer ports etc
+- e2e test cases must cover various combinations of `workloadSelector` in `appliedTo` and ingress/egress rules
+- e2e test cases must cover various combinations of `namespaceSelector` in `appliedTo`
+- e2e test cases must cover various combinations of `namespaces` in ingress/egress rules
+  - Ensure that namespace matching strategies work as expected
+- Add unit tests to test the validation logic which shall be introduced for cluster-scoped policy resources
+  - Ensure that `self` field cannot be set along with `selector` within `namespaces`
+  - Test cases for fields which are shared with NetworkPolicy, like `endPort` etc.
+  - Ensure no two policies can be set with same `priority`
+- Ensure that only administrators or assigned roles can create/update/delete cluster-scoped policy resources
+
+### Graduation Criteria
+
+#### Alpha to Beta Graduation
+
+- Gather feedback from developers and surveys
+- At least 2 CNI provider must provide the implementation for the complete set
+  of alpha features
+- Evaluate "future work" items based on feedback from community
+- Ensure extensibility of adding new fields. i.e. adding new fields do not "fail-open" traffic for older clients
+
+#### Beta to GA Graduation
+
+- At least 4 CNI providers must provide the implementation for the complete set
+  of beta features
+- More rigorous forms of testing
+  — e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+- Completion of all accepted "future work" items
+
+### Upgrade / Downgrade Strategy
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to maintain previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade, in order to make use of the enhancement?
+-->
+
+#### Upgrade considerations
+
+As such, the cluster-scoped policy resources are new and shall not exist prior
+to upgrading to a new version. Thus, there is no direct impact on upgrades.
+
+#### Downgrade considerations
+
+Downgrading to a version which no longer supports cluster-scoped policy APIs
+must ensure that appropriate security rules are created to mimick the cluster-scoped
+policy rules by other means, such that no unintended traffic is allowed.
```

Comment:
And also that all _intended_ traffic _is_ allowed. eg, if there was a CNP that Allowed access to CoreDNS, they would need to make sure that everyone still had access without it